src/medLog/medLogger.cpp

```diff
@@ -0,0 +1,117 @@
+#include "medLogger.h"
+
+#include <iostream>
+
+medLogger* medLogger::singleton;
+bool medLogger::logAccessFlag = false;
+
+medLogger& medLogger::instance()
+{
+    return *singleton;
```

Comment:
(my comment BTW is another type of fcolloting: too much explanation which makes it hard to understand).